lib/api.js

```diff
@@ -0,0 +1,306 @@
+'use strict';
+
+var bitcore = require('bitcore-lib');
+var buffer = require('buffer');
+var aesjs = require('aes-js');
+var scrypt = require ('scrypt-async');
+var base64 = require('base64-js');
+var request = require('browser-request');
+
+//Constants' definitions
+
+var BASE_URL = 'http://localhost:10000/jointescrow';
+var WEBSOCKET_ENDPOINT = '/common-events';
+var LOGIN_ENDPOINT = '/login/';
+var REGISTER_ENDPOINT = '/register/';
+var CREATE_CONTRACT_ENDPOINT = '/create-contract/';
+var ADD_COUNTERPARTY_ENDPOINT = '/add-counterparty/';
+var GET_CONTRACT_ENDPOINT = '/get-contract/';
+var PAYLOAD_ENDPOINT = '/key';
+var GEAR_AUTH_ENDPOINT = '/auth/gear';
+var JE_AUTH_INFO = '/auth';
+var JE_INFO = '/info';
+var GET_CONTRACTS_LIST_ENDPOINT = '/contracts/limit/';
+
+var SALT = 'sd235o2i23v52oip426ophfaopia234poihgpsodopihstgo';
+
+
+var PRIV_KEY_NAME = 'privkey';
+
+/**
+ * @desc MyceliumContracts constructor.
+ *
+ * @param {Object} opts
+ * @constructor
+ */
+function MyceliumContracts(opts) {
+  opts = opts || {};
+
+  this.verbose = !!opts.verbose;
+  this.request = opts.request || request;
+  this.baseUrl = opts.baseUrl || BASE_URL;
+  this.timeout = opts.timeout || 50000; //Write comments for each values!!
+
+  this.initNotifications();
+};
+
+
+MyceliumContracts.prototype.initNotifications = function() {
+
+  var ws = new WebSocket("ws://localhost:10000/jointescrow" + WEBSOCKET_ENDPOINT);
+
+  ws.onmessage = function(event) {
+      console.log(Date.now() + " " + event.data);
+  };  
+}
+
+/**
+ * Does an HTTP request
+ * @private
+ *
+ * @param {Object} method
+ * @param {String} url
+ * @param {Object} args
+ * @param {Callback} callback
+ */
+MyceliumContracts.prototype._ajax = function(args) {
+
+  if (!args.withCredentials)
+    args.withCredentials = true;
+
+  if (!args.timeout)
+    args.timeout = this.timeout;
+
+  this.request(args, function(err, res, body) {    
+  
+    if (!res) {
+      return args.callback({message : "Connection error"});
+    }
+
+    if (res.statusCode !== 200) {
+      if (res.statusCode === 404)
+        return args.callback({code : 404, message: "Not found"});
+
+      if (!res.statusCode)
+        return args.callback({message : "Connection error"});
+
+      if (!body)
+        return args.callback({code : res.statusCode, message : "HTTP error"});
+
+      return args.callback({code : res.statusCode, message : 'Error in body'});
+    }
```

Comment:
What about all 2xx statuses? And 304 Not Modified for api? Cat we get error message from response or error object?